HISTORY.md

```diff
@@ -1,3 +1,7 @@
+### mlpack 3.0.5
+###### 2019-01-02
+  * Fix DBSCAN isn't using PointSelectionPolicy issue (#1625).
```

Comment:
I know this is a minor comment, but can we make this say something more active, like `Change DBSCAN to use PointSelectionPolicy and add RandomSelectionPolicy (#1625)`.  Also we can change the date to `????-??-??`, since I am not sure when the release will be. :)

Reply:
The style you used is great overall, so thank you for adhering to the style guidelines. :)